walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -121,6 +124,66 @@ public void everyAttendeeIsConsidered() {
     Assert.assertEquals(expected, actual);
   }
 
+  @Test
+  public void everyAttendeeIsConsideredWithOptionalAllDay() {
+    // Same as everyAttendeeIsConsidered() but with an extra optional attendee who has an all-day
+    // event. Since they are never available, this should return the same three time slots.
+    //
+    // Events  :       |--A--|     |--B--|
+    //           |==============C==============|
+    // Day     : |-----------------------------|
+    // Options : |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.WHOLE_DAY, Arrays.asList(PERSON_C)));
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
```

Comment:
FYI: Whenever you pass in an argument to a function in a test like this without creating a variable for it, you should consider whether you want to include the parameter name so that the code is unambiguous to the reader. In this case, it seems pretty clear what the parameters are to MeetingRequest. Just for demonstration, though, this is what it would look like if you wanted to include extra information (which you should in cases where it's less obvious what the parameters are):

```new MeetingRequest(/*attendees=*/Arrays.asList(PERSON_A, PERSON_B), /*duration=*/DURATION_30_MINUTES)```